packages/react-components/react-drawer/library/src/components/InlineDrawer/useInlineDrawer.ts

```diff
@@ -16,26 +24,44 @@ import { useDrawerDefaultProps } from '../../shared/useDrawerDefaultProps';
  */
 export const useInlineDrawer_unstable = (props: InlineDrawerProps, ref: React.Ref<HTMLElement>): InlineDrawerState => {
   const { size, position, open } = useDrawerDefaultProps(props);
-  const { separator = false } = props;
-
-  const motion = useMotion<HTMLElement>(open);
+  const { separator = false, surfaceMotion } = props;
 
-  return {
+  const state: InlineDrawerState = {
     components: {
       root: 'div',
+      // casting from internal type that has required properties
+      // to external type that only has optional properties
+      // converting to unknown first as both Function component signatures are not compatible
+      surfaceMotion: InlineDrawerMotion as unknown as React.FC<SurfaceMotionSlotProps>,
     },
 
     root: slot.always(
       getIntrinsicElementProps('div', {
         ...props,
-        ref: useMergedRefs(ref, motion.ref),
+        ref,
       }),
       { elementType: 'div' },
     ),
 
-    size,
+    open,
     position,
+    size,
     separator,
-    motion,
+    surfaceMotion: presenceMotionSlot<DrawerMotionParams>(surfaceMotion, {
+      elementType: InlineDrawerMotion,
+      defaultProps: {
+        position,
+        size,
+        visible: open,
+        unmountOnExit: true,
+      },
+    }),
+
+    // Deprecated props
+    motion: STATIC_MOTION,
   };
+  if (typeof state.components.surfaceMotion === 'function') {
+    state.components.surfaceMotion.defaultProps;
```

Comment:
Nothing being called here. Is this correct?

Reply:
Yes, it's garbage, no idea why it's there, removed...